qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -130,6 +129,36 @@ def __json_encode__(self):
         return self.__getstate__()
 
 
+class FigureData:
+    """Wrapper for figures and figure metadata"""
+
+    def __init__(self, figure, name=None, metadata=None):
+        """Creates a new figure data object"""
+        self.figure = figure
+        self._name = name
+        self.metadata = metadata if metadata is not None else {}
+
+    # name is read only
+    @property
+    def name(self):
+        """The name of the figure"""
+        return self._name
+
+    def copy(self, new_name: Optional[str] = None):
+        """Creates a deep copy of the figure data"""
+        name = new_name if new_name is not None else self.name
+        return FigureData(figure=self.figure, name=name, metadata=copy.deepcopy(self.metadata))
```

Comment:
This isn't a deep copy if the figure isn't also deep copied. If you want a deepcopy why not just do `copy.deepcopy(self)` ?

Reply:
Just tried it and deep copying the figure causes matplotlib errors.